mcpscanner/core/static_analysis/native_analyzer.py

```
@@ -3730,6 +3756,9 @@ def _ts_extract_function_context(
 
         # Extract docstring/JSDoc from AST
         docstring = self._ts_extract_docstring(node)
+        anno_description = self._parse_description_from_annotations(node)
+        if anno_description:
+            docstring = anno_description if not docstring else anno_description
```

Ruff (0.15.20)

[warning] Useless `if`-`else` condition

(RUF034)